packages/directory/addon/components/dir-asset-row-actions.js

```diff
@@ -3,50 +3,44 @@
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  *
  * Usage:
- * {{dir-asset-row-action
- *   value=assetModel
- *   row=rowObject
- * }}
+ * <DirAssetRowActions
+ *   @value={{assetModel}}
+ *   @row={{rowObject}}
+ * />
  */
 import Component from '@ember/component';
 import layout from '../templates/components/dir-asset-row-actions';
 import { computed, get } from '@ember/object';
+import { layout as templateLayout, classNames } from '@ember-decorators/component';
 
-export default Component.extend({
-  layout,
-
-  /**
-   * @property {Array} classNames
-   */
-  classNames: ['dir-asset-row-actions'],
-
+@templateLayout(layout)
+@classNames('dir-asset-row-actions')
+class DirAssetRowActions extends Component {
   /**
    * @property {String} type - the type of the asset
    */
-  type: computed('value', function() {
-    let { value } = this;
-
+  @computed('value')
+  get type() {
+    const { value } = this;
     return value ? value.constructor.modelName : null;
-  }),
+  }
 
   /**
    * @property {String} rowElement - selector for the row containing this
    */
-  rowElement: computed('value', 'row.rowId', function() {
-    let rowId = get(this, 'row.rowId');
-
+  @computed('value', 'row.rowId')
+  get rowElement() {
+    const rowId = get(this, 'row.rowId');
     return `[data-row-id=${rowId}]`;
-  }),
+  }
 
   /**
    * @property {Object} assetActionComponents - action list component for each asset type
    */
-  assetActionComponents: computed(() => {
-    const components = {
-      report: 'navi-action-list',
-      dashboard: 'dashboard-action-list'
-    };
+  assetActionComponents = {
```

Comment:
Is the state leaking issue fixed in native classes?

Reply:
Yup, see [Class Fields vs. `extend()`](https://blog.emberjs.com/2019/01/26/emberjs-native-class-update-2019-edition.html)